ocs_ci/deployment/provider_client/storage_client_deployment.py

```diff
@@ -0,0 +1,630 @@
+"""
+This module provides installation of ODF in provider mode and storage-client creation
+on the hosting cluster.
+"""
+import atexit
```

Comment:
I think we should not use atexit because this is a pytest framework, it has tearDown and hooks, such as @pytest.hookimpl(trylast=True)

atexit can lead to issues and unexpected behavior such as causing logging errors and tracebacks to be printed at the end of the test session, polluting the output https://github.com/pytest-dev/pytest/issues/5282

cc @dahorak @petr-balogh

Reply:
I will address this in future as its not an immediate requirement.

Reply:
Have removed in latest commit